### ISSUE TITLE
updateAt() proposal

### ISSUE THREAD

**Issue by romaincazier** [tweenjs/tween.js#391]
Hi,

For a project I'm working on, I needed to get an output value based on the scroll, so I wrote down a function that would allow me to get a it using your library. Maybe it could be cool to add ? I'm not that familiar with github, so I'm posting this here : 

`updateAt: function updateAt(time) {

	// We don't want to mess with a playing Tween
	if(this._isPlaying) return;
		
	// Since we didn't call start() we need to populate this._valuesStart (only once)
	if(Object.getOwnPropertyNames(this._valuesStart).length == 0) {
		for (var property in this._valuesEnd) {
			if(this._object[property] === undefined) {
				continue;
			}
				
			this._valuesStart[property] = this._object[property];
		}
	}
				

	this._startTime = 0;

	time = time < 0 ? 0 : time > this._duration ? this._duration : time;
	this.update(time);

}`

And here a simple example using it (for the sake of simplicity I'm not optimizing) : 

`var manualTween = new TWEEN.Tween({x: 0})
	.to({x: 100}, 1)
	.onUpdate((object) => {
		// do whatever
		console.log(object.x);
	})
	.easing(TWEEN.Easing.Exponential.InOut);

window.addEventListener('scroll', () => {
	var t = document.body.scrollTop / window.innerHeight;
	manualTween.updateAt(t);
});`

I don't know about the naming, or the way I wrote the function so I'm opening this discussion :)

**Comment by mikebolt** [comment]
So this lets you control a tween using a value that might decrease, such as the page scroll height?

Did you try using `tween.start(0)`?

**Comment by romaincazier** [comment]
The idea is mostly about getting an output for a specific value. 

The problem of `tween.start()` is that it's adding the tween to the group update and thus playing the animation. Think of it as a paused tween which I move to whatever time I want to get a specific value. Here I'm using the scroll but I imagine it could be useful in other cases.

**Comment by romaincazier** [comment]
Quick example : https://codepen.io/anon/pen/bYGEwB

Hope this illustrates better

**Comment by mikebolt** [comment]
How does this interact with the `update` method? Will `update` behave the same as it did before if `updateAt` is called before a tween?

**Comment by romaincazier** [comment]
Yes, it's not destructive or anything. If you call `start()` the tween will behave normally.

**Comment by romaincazier** [comment]
I updated the example : https://codepen.io/anon/pen/bYGEwB

If you move the mouse, and then stop and click, you see that it plays the animation normally.

**Comment by martikaljuve** [comment]
If you only want to map one number to another number, you could just use the easings in TWEEN.Easing directly, e.g. in your example:

```js
document.body.addEventListener('mousemove', function(e) {
	var relX = e.pageX / window.innerWidth;
	var scale = window.innerWidth - 100;

	var x = TWEEN.Easing.Exponential.InOut(relX) * scale;

	square.style.transform = `translate(-50%, -50%) translateX(${x}px)`;
});
```

**Comment by romaincazier** [comment]
@martikaljuve actually yeah I saw afterwards that I could just do it this way... I don't know why but it made sense to me to do it the way I described in my first post. Anyway, you can close the thread if you want, we can considered it as closed :)
